ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogContainer.tsx

```diff
@@ -201,21 +200,25 @@ export function BrowseFilesDialogContainer(props: BrowseFilesDialogContainerProp
 	const onCloseButtonClick = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => onClose(e, null);
 
 	const onUpload = () => {
+		const dialogId = nanoid();
 		dispatch(
-			showSingleFileUploadDialog({
-				site,
-				path: currentPath,
-				fileTypes: mimeTypes,
-				onClose: closeSingleFileUploadDialog(),
-				onUploadComplete: batchActions([closeSingleFileUploadDialog(), dispatchDOMEvent({ id: 'imageUploaded' })])
+			pushDialog({
+				id: dialogId,
+				component: 'craftercms.components.SingleFileUploadDialog',
+				props: {
+					site,
+					path: currentPath,
+					fileTypes: mimeTypes,
+					onClose: () => dispatch(popDialog({ id: dialogId })),
+					onUploadComplete: () => {
+						dispatch(popDialog({ id: dialogId }));
+						setTimeout(() => {
+							fetchItems();
+						}, 2000);
```

Comment:
Is this timeout is still a necessity?

Reply:
yeah it still sometimes fails to fetch up to date items